gmeta/codegen/src/lib.rs

```diff
@@ -1,518 +1,503 @@
-            ))
+/// Generates metawasm functions.
+///
+/// An example of the expected structure:
+/// ```
+/// use gstd::prelude::*;
+///
+/// #[derive(Decode, Encode, TypeInfo)]
+/// pub struct StateType;
+///
+/// #[derive(Encode, TypeInfo)]
+/// pub struct SomeReturnType;
+///
+/// #[derive(Decode, TypeInfo)]
+/// pub struct SomeArg;
+///
+/// #[gmeta::metawasm]
+/// pub mod metafuncs {
+///     pub type State = StateType;
+///
+///     /// Documentation...
+///     pub fn some_function(_: State) -> SomeReturnType {
+///         unimplemented!()
+///     }
+///
+///     pub fn another_function_but_with_arg(mut _state: State, _arg: SomeArg) -> State {
+///         unimplemented!()
+///     }
+///
+///     /// DOCSdocsDoCsdOcSDoCS...
+///     pub fn function_with_multiple_args(
+///         _state: State,
+///         mut _arg1: SomeArg,
+///         _arg2: u16,
+///         mut _arg3: u32,
+///     ) -> SomeReturnType {
+///         unimplemented!()
+///     }
+/// }
+/// # fn main() {}
+/// ```
+///
+/// # Syntax
+///
+/// - This attribute **must** be used on the `pub`lic `mod` container with the
+/// `metafuncs` identifier.
+/// - The first item in the module **must** be a `pub`lic `type` alias with the
+/// `State` identifier. The type for which `State` will be an alias **must**
+/// implement [`Decode`] trait.
+///
+/// Usually the state type should be imported from the implemented associative
+/// [`Metadata::State`](../gmeta/trait.Metadata.html#associatedtype.State) type
+/// from the contract's `io` crate.
+///
+/// - The rest of items **must** be `pub`lic functions.
+/// - The first argument's type of metafunctions **must** be `State`.
+///
+/// In addition to the mandatory first argument, functions can have additional
+/// ones.
+///
+/// - The maximum amount of additional arguments is 18 due restrictions of the
+/// SCALE codec.
+/// - All additional arguments **must** implement the [`Decode`] &
+/// [`TypeInfo`] traits.
+/// - A function **mustn't** return `()` or nothing.
+/// - A returned type **must** implement the
+/// [`Encode`](../gmeta/trait.Encode.html) & [`TypeInfo`] traits.
+///
+/// [`Decode`]: ../gmeta/trait.Decode.html
+/// [`TypeInfo`]: ../gmeta/trait.TypeInfo.html
+///
+/// # Expansion result
+///
+/// This attribute doesn't change the `metafuncs` module and items inside, but
+/// adds `use super::*;` inside the module because, in most cases, it'll be
+/// useful for importing items from an upper namespace. So every item in the
+/// same namespace where the module is located is accessible inside it.
+///
+/// The rest of the magic happens in the another generated private `extern`
+/// module. It registers all metawasm functions, their arguments & return types,
+/// and generates extern functions with the same names. Later, they can be
+/// called from a metaWASM binary inside a blockchain.
+///
+/// **Important note**: although metafunctions can take more than 1 additional
```

Comment:
clarify please that this explanation is about underthehood behaviour, not the syntax below macro